tools/dump_tool/dump_tool.c

```diff
@@ -244,6 +244,50 @@ static int fsdump_recv(int dev_fd)
 	return 0;
 }
 
+static int extfsdump_recv(int dev_fd)
+{
+        int ret;
+        uint32_t ext_smartfs_size;
+        char ext_fs_bin[BINFILE_NAME_SIZE] = { '\0' };
+        FILE *fs_fp;
+
+        /* Receive Userfs flash partition size */
+        ret = b_read(dev_fd, (uint8_t *)&ext_smartfs_size, 4);
+        if (ret != 4) {
+                printf("Receiving Userfs size failed, ret = %d\n", ret);
+                return -1;
+        }
+
+	if (ext_smartfs_size == 0) {
+		printf("No external userfs partition found. Please check config\n");
+		return 0;
+	}
+
+        printf("\n=========================================================================\n");
+        printf("External filesystem size = %d\n", ext_smartfs_size);
```

Comment:
`ext_smartfs_size` is unsigned type, so please print its size with `%u` instead of `%d`.

Reply:
Updated